packages/apollo-graphql/src/schema/buildSchemaFromSDL.ts

```diff
@@ -216,6 +221,38 @@ export function addResolversToSchema(
       }
     }
 
+    if (isScalarType(type)) {
+      for (const fn in fieldConfigs) {
+        (type as any)[fn] = (fieldConfigs as any)[fn];
+      }
+    }
+
+    if (isEnumType(type)) {
+      const values = type.getValues();
+      const newValues = {};
+      values.forEach(value => {
+        const newValue = (fieldConfigs as any)[value.name] || value.name;
+        (newValues as any)[value.name] = {
+          value: newValue,
+          deprecationReason: value.deprecationReason,
+          description: value.description,
+          astNode: value.astNode
+        };
+      });
+
+      // In place updating hack to get around pulling in the full
+      // schema walking and immutable updating machinery from graphql-tools
+      Object.assign(
+        type,
+        new GraphQLEnumType({
+          name: type.name,
```

Comment:
Small nit: we can replace the "cloned" keys `name, description, astNode` with `...type.toConfig()`, which is a relatively new addition for convenience when we transform types, schema, etc.

Reply:
Bit confused, `type.toConfig()` doesn't provide `deprecationReason`, and `value` doesn't have a `.toConfig()`.

Reply:
The values work up above is fine, to be more clear I'm suggesting this for building the new `GraphQLEnumType`. There's no `deprecationReason` on an enum type itself, just on its values:
https://github.com/graphql/graphql-js/blob/master/src/type/definition.js#L1202-L1224

```js
Object.assign(
  type,
  new GraphQLEnumType({
    ...type.toConfig(),
    values: newValues
  })
);

Reply:
Ah that makes sense, thanks!